### PR TITLE
Use Blosc special values in empty arrays

### PR DESCRIPTION
**Pull request by aleixalcacer**
- [x] Use the Blosc machinery for special values to create empty arrays.
- [x] Allow the users to update array elements.

### CODE REVIEW

**Review by oscargm98 — APPROVED** [review]

**Review by FrancescAlted — COMMENTED** [review]
Looks really good!  I'd just suggest to add some benchmarks comparing the performance with the previous version.  Add the benchs to a bench/ dir, and put the output for the previous version and this one as a comment in this PR.  Then, just add an entry in the RELEASE_NOTES where you refer to this PR, so that the benchmarks can be visible to anybody.

**Comment by aleixalcacer** [comment]
The results of the new benchmark file (bench/bench_get_slice.c) are:

- master
  ```
  from_buffer: 3.3 s
  get_slice: 0.79 s
  ```
- this PR
  ```
  from_buffer: 2.4 s
  get_slice: 0.59 s
  ```

As we can see, there exists a performance improvement!

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -4,7 +4,17 @@ Release notes
 Changes from 0.4.0 to 0.4.1
 ---------------------------
 
-XXX version-specific blurb XXX
+* Redesign the caterva guts and perform a code refactorization in order
+  to simplify the code. This includes API renaming. A performance improvement
+  is obtained as can be seen in
```
Comment (FrancescAlted) [inline RELEASE_NOTES.rst]: You missed the reference to PR

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -4,7 +4,17 @@ Release notes
...
+
+* Implement a set_slice for arrays backed by Blosc. This allows users to
+  update the values in the array whenever they want.
```
Comment (FrancescAlted) [inline RELEASE_NOTES.rst]: whenever and *wherever* :-)

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -4,7 +4,17 @@ Release notes
...
+
+* Implement constructors (empty, zeros, full) using the special-values
+  features introduced in Blosc.
```
Comment (FrancescAlted) [inline RELEASE_NOTES.rst:15]: To be consistent with C-Blosc2, I'd add the NaN constructor too, but that can come later.

**Review by FrancescAlted — APPROVED** [review]
Looks very good to me!  I just have some comments.

**Pull request merged**